src/pkg/cli/client/byoc/do/byoc.go

```diff
@@ -219,28 +224,24 @@ func (b *ByocDo) BootstrapCommand(ctx context.Context, req client.BootstrapComma
 	if err != nil {
 		return "", err
 	}
-	etag := pkg.RandomID()
 
+	etag := pkg.RandomID()
+	b.lastCdEtag = etag
 	return etag, nil
 }
 
 func (b *ByocDo) BootstrapList(ctx context.Context) ([]string, error) {
-	// Use DO api to query which apps (or projects) exist based on defang constant
-
-	var projectList []string
-
-	projects, _, err := b.client.Projects.List(ctx, &godo.ListOptions{})
+	s3client, err := b.driver.CreateS3Client()
 	if err != nil {
 		return nil, err
 	}
 
-	for _, project := range projects {
-		if strings.Contains(project.Name, "Defang") {
-			projectList = append(projectList, project.Name)
-		}
+	bucketName, err := b.driver.GetBucketName(ctx, s3client)
+	if bucketName == "" {
+		return nil, err
 	}
 
-	return projectList, nil
+	return awsbyoc.ListPulumiStacks(ctx, s3client, bucketName)
```

Comment:
Should we move this to some kind of new common module instead of referencing aws code from do?

Reply:
I started looking into this but it turned into a whole refactoring because of other devs, so I won't do it in here.